### PR TITLE
Merged Card models with Card node.

### PR DESCRIPTION
**Pull request by emmettirl**
Adapted Parsers, and Deck/Combo Models make use of node. 

Commented out GDscript card object which was temporarily interfering

### CODE REVIEW

File: scripts/battle/Battle.cs [hunk 1]
```diff
@@ -4,112 +4,110 @@
...
+		selectedEnemyIndex = selectedEnemyIndex != enemyIndex ? enemyIndex : -1;
+	}
+
+	private void OnPlayButtonPressed() {
+		PlaySelectedCard();
+	}
+
+	private void OnDeckPressed() {
+		hand.AddCards(deck.DrawCard());
+	}
```
Comment (VonJellybean) [inline scripts/battle/Battle.cs:112]: I'm using 4 space indentation. Unless we make a decision this is going to be hell.

File: scripts/battle/card/AttackCard.cs
Comment (VonJellybean) [inline scripts/battle/card/AttackCard.cs:1]: What does this card type add that would not be covered by the default behaviour of a card being played? What does it add?

File: scripts/battle/card/Card.cs [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+public partial class Card : Node2D {
+	[Signal]
+	public delegate void CardSelectedEventHandler(Card card);
+	public Color color { get; set; }
+	public Button selectButton;
+	
+	
+	public string Id { get; set;} // card_id
+
+	public TargetEnum Target { get; set;}
```
Comment (VonJellybean) [inline scripts/battle/card/Card.cs:13]: This is handled by the Battle code and doesn't need to be saved by the card.

File: scripts/customresource/GDCard.gd
Comment (VonJellybean) [inline scripts/customresource/GDCard.gd:1]: I don't think these GDscript files are needed as we've translated the functionality to C#

File: scripts/battle/Battle.cs (hunk 1 above)
Comment (emmettirl) [reply scripts/battle/Battle.cs:112]: 4 space it is haha, Rider has been doing that automatically for me. 
Will update my IDE later so it deafults to it.

File: scripts/battle/card/AttackCard.cs
Comment (emmettirl) [reply scripts/battle/card/AttackCard.cs:1]: There are effects based on card type. 
Different types have interactions with the position status. 

Attack/Block cards remove status, and clear combo stack. 
I think the others have effects, but I do not recall them off hand. For a future sprint

File: scripts/customresource/GDCard.gd
Comment (emmettirl) [reply scripts/customresource/GDCard.gd:1]: Removed these in my next push after review

File: scripts/battle/card/Card.cs (hunk 2 above)
Comment (emmettirl) [reply scripts/battle/card/Card.cs]: done

File: scripts/battle/Battle.cs (hunk 1 above)
Comment (VonJellybean) [reply scripts/battle/Battle.cs:112]: Can you change the indentation now to prevent and merge conflicts?

Comment (emmettirl) [reply scripts/battle/Battle.cs]: Also pushed an update to return this to 4space

File: scripts/battle/card/Card.cs
Comment (VonJellybean) [inline scripts/battle/card/Card.cs:1]: Whitespace

**Pull request merged**